model/src/channel/message/allowed_mentions/builder.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    channel::message::allowed_mentions::{AllowedMentions, ParseTypes},
+    id::{RoleId, UserId},
+};
+
+#[derive(Clone, Default, Debug, Eq, PartialEq)]
+pub struct AllowedMentionsBuilder(AllowedMentions);
+
+impl AllowedMentionsBuilder {
+    /// Create a new [`AllowedMentionsBuilder`].
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Allow parsing of `@everyone`.
+    pub fn everyone(mut self) -> Self {
+        self.0.parse.push(ParseTypes::Everyone);
+
+        self
+    }
+
+    /// When replying, whether to mention the target user.
+    pub fn replied_user(mut self) -> Self {
+        self.0.replied_user = true;
+
+        self
+    }
+
+    /// Allow parsing of all roles.
+    pub fn roles(mut self) -> Self {
+        self.0.parse.push(ParseTypes::Roles);
+
+        self
+    }
+
+    /// Allow parsing of specific [`RoleId`]s.
+    ///
+    /// Note: If called, this will override the effect of [`roles`].
```

Comment:
My first questions were "What does this mean? And why does it do that?" but then I realized. This should be expanded in some way.